cheroot/connections.py

```diff
@@ -282,12 +266,27 @@ def _from_server_socket(self, server_socket):
 
     def close(self):
         """Close all monitored connections."""
-        for conn in self.connections[:]:
+        for conn in self._readable_conns:
             conn.close()
-        self.connections = []
+        self._readable_conns.clear()
+
+        for _, key in self._selector.get_map().items():
+            if key.data != self.server:  # server closes its own socket
+                key.data.socket.close()
+
+        self._selector.close()
+
+    def _num_connections(self):
```

Comment:
This should be a property semantically
```suggestion
    @property
    def _num_connections(self):
```